awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -212,46 +190,41 @@ def make_request(self, method, endpoint, *args, **kwargs):
             # If we have a oauth token, we just use a bearer header
             headers['Authorization'] = 'Bearer {0}'.format(self.oauth_token)
 
-        # Update the URL path with the endpoint
-        self.url = self.url._replace(path=endpoint)
```

Comment:
@john-westcott-iv the variable `self.url` is defined in the parent class `TowerModule`. I'm fine to have the subclasses reference it (like other config params), but I don't want them to _modify_ it. That makes state hard to track. By only working with a local `url` instead of `self.url` we get rid of the `finally:` block to reset it to the prior value. It's easier to just never modify instance data so we don't have to worry about state.